src/Sobol.jl

```diff
@@ -1,67 +1,93 @@
 module Sobol
 
-import NLopt.libnlopt
-import Base: ndims, length, skip, next, start, done, show, writemime
+import Base: ndims, skip, next, start, done, show, writemime
 export SobolSeq, ScaledSobolSeq, next!
 
-type SobolSeq{N}
-    p::Ptr{Void} # nlopt_sobol pointer
-    function SobolSeq()
-        (N < 1 || N > 1111) && error("invalid Sobol dimension")
-        p = ccall((:nlopt_sobol_create,libnlopt), Ptr{Void}, (Cuint,), N)
-        p == C_NULL && error("error initializing Sobol sequence")
-        s = new(p)
-        finalizer(s, destroy)
-        s
-    end
+include("soboldata.jl") #loads `sobol_a` and `sobol_minit`
+
+type SobolSeq
+    sdim::Int #dimension of sequence being generated
+    m::Array{UInt,2} #array of size (sdim, WORD_SIZE)
+    x::Array{UInt,1} #previous x = x_n, array of length sdim
+    b::Array{UInt,1} #position of fixed point in x[i] is after bit b[i]
+    n::UInt #number of x's generated so far
 end
-SobolSeq(N::Integer) = SobolSeq{N}()
 
-destroy(s::SobolSeq) = 
-    ccall((:nlopt_sobol_destroy,libnlopt), Void, (Ptr{Void},), s.p)
+ndims(s::SobolSeq) = s.sdim
 
-ndims{N}(s::SobolSeq{N}) = N
+function SobolSeq(N::Integer)
```

Comment:
In principle there is no reason not to support `N == 0`.  This is just the trivial case, of course, where there is nothing to generate.

(A nice thing about using `N` as a type parameter is that then we can have different methods for different `N`.  e.g. `next(s::SobolSeq{0}, x)` can just be a two-line function to check the length of `x` and then return `x`